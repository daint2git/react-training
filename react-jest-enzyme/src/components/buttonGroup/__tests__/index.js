import ButtonGroup from '../'

describe('ButtonGroup', () => {
  test('using shallow', () => {
    const shallowed = shallow(<ButtonGroup />)

    expect(shallowed).toMatchSnapshot()
  })

  test('using mount', () => {
    const mounted = mount(<ButtonGroup />)

    expect(mounted).toMatchSnapshot()
  })

  test('using render', () => {
    const rendered = render(<ButtonGroup />)

    expect(rendered).toMatchSnapshot()
  })

  test('using react-test-renderer', () => {
    const tree = renderer.create(<ButtonGroup />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
