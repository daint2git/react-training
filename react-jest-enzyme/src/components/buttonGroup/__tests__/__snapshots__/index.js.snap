// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup using mount 1`] = `
<styled.div>
  <div
    className="sc-bwzfXH jIEWCi"
  >
    <styled.button
      className="class"
      type="button"
    >
      <button
        className="class sc-bdVaJa ihDGdR"
        type="button"
      >
        <strong>
          Hello!
        </strong>
      </button>
    </styled.button>
    <styled.button
      className="class1"
      type="button"
    >
      <button
        className="class1 sc-bdVaJa ihDGdR"
        type="button"
      >
        <strong>
          Hello2!
        </strong>
      </button>
    </styled.button>
  </div>
</styled.div>
`;

exports[`ButtonGroup using react-test-renderer 1`] = `
<div
  className="sc-bwzfXH jIEWCi"
>
  <button
    className="class sc-bdVaJa ihDGdR"
    type="button"
  >
    <strong>
      Hello!
    </strong>
  </button>
  <button
    className="class1 sc-bdVaJa ihDGdR"
    type="button"
  >
    <strong>
      Hello2!
    </strong>
  </button>
</div>
`;

exports[`ButtonGroup using render 1`] = `
<div
  class="sc-bwzfXH jIEWCi"
>
  <button
    class="class sc-bdVaJa ihDGdR"
    type="button"
  >
    <strong>
      Hello!
    </strong>
  </button>
  <button
    class="class1 sc-bdVaJa ihDGdR"
    type="button"
  >
    <strong>
      Hello2!
    </strong>
  </button>
</div>
`;

exports[`ButtonGroup using shallow 1`] = `
<styled.div>
  <Button
    className="class"
  >
    <strong>
      Hello!
    </strong>
  </Button>
  <Button
    className="class1"
  >
    <strong>
      Hello2!
    </strong>
  </Button>
</styled.div>
`;
